Canvas/GradesWidget/GradesTodayWidgetViewController.swift

```diff
@@ -80,135 +64,55 @@ func isDeviceLocked() -> Bool {
 
 class GradesTodayWidgetViewController: UIViewController {
 
-    var courses: [Course] = [] {
-        didSet {
-            if courses.count > 0 {
-                preferredContentSize = CGSize(width: 0, height: CGFloat(courses.count) * tableView.estimatedRowHeight)
-            } else {
-                self.showError(GradesWidgetError.noFavoritedCourses)
-                preferredContentSize = CGSize(width: 0, height: tableView.estimatedRowHeight)
-            }
-        }
-    }
-    var colors: CustomColors?
+    lazy var presenter: GradesTodayWidgetPresenter = {
+        return GradesTodayWidgetPresenter(view: self)
+    }()
+
     var error: Error?
-    var colorRetries: Int = 0
-    var courseRetries: Int = 0
 
-    var client: CKIClient?
+    @IBOutlet var tableView: UITableView!
+
+    override func viewDidLoad() {
+        view.backgroundColor = UIColor.clear
+        extensionContext?.widgetLargestAvailableDisplayMode = .expanded
 
-    lazy var tableView: UITableView = {
-        let tableView = UITableView(frame: .zero, style: .plain)
         tableView.backgroundColor = UIColor.clear
         tableView.translatesAutoresizingMaskIntoConstraints = false
         tableView.dataSource = self
         tableView.delegate = self
         tableView.showsVerticalScrollIndicator = false
         tableView.estimatedRowHeight = COURSE_ROW_HEIGHT
         tableView.rowHeight = UITableView.automaticDimension
-        tableView.tableFooterView = UIView(frame: .zero)
 
-        // Cells
         let nib = UINib(nibName: String(describing: GradeWidgetCell.self), bundle: nil)
         tableView.register(nib, forCellReuseIdentifier: "cell")
 
-        return tableView
-    }()
-
-    override func didReceiveMemoryWarning() {
```

Comment:
we may want to leave this `didReceiveMemoryWarning` method.  The widgets don't get as much memory as apps do.  I think this was pivotal to finding a bug if i remember right, @armstrongnate can you confirm how you stress tested the widget in the past?

Reply:
That's right. This warning is especially relevant for today widgets.